### PR TITLE
Locale scaffolding

### PR DESCRIPTION
**Pull request by KludgeKML**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What

Tidy up the locale files, add empty files for the additional 64 locales (frontend at the moment only supports en/cy) and add a task that will allow us to copy across all translations of a specific localisation tree.

## Why

https://trello.com/c/woZwSv9D/363-create-translation-scaffolding-for-frontend, [Jira issue PNP-8549](https://gov-uk.atlassian.net/browse/PNP-8549)

## How

Use rails_translation_manager / i18n-tasks where possible (but they don't support this exact use case, so some workarounds)

### CODE REVIEW

File: lib/tasks/consolidation.rake [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# We don't care much about the coverage of scaffolding files that are just used on the desktop during
+# consolidation and are going to be removed.
+# :nocov:
+namespace :consolidation do
+  desc "Copy a locale key for all locales from government-frontend to frontend"
```
[the remark this answers is not in the document]
Comment (leenagupte) [reply lib/tasks/consolidation.rake:5]: Also, if you're using zsh like I am, you'll need to escape the brackets for the task to work, e.g

```
rake consolidation:copy_translation\[travel_advice\]
```

Otherwise you get an error:

```
zsh: no matches found: consolidation:copy_translation[travel_advice]
```

Perhaps this too could be added to the task description. Perhaps as a usage statement?

Apparently `rake -D` show the whole [description](https://ruby.github.io/rake/Rake/DSL.html#method-i-desc) if it's over multiple lines.

**Review by leenagupte — COMMENTED** [review]
I've tried this out locally, and it works like a charm.

I've added recommendations for updating the task description inline.

There aren't any tests for this which isn't ideal, but as it's only to help local development and the changes can be easily unchanged, I think in this case it's ok.

File: config/locales/cy.yml [hunk 2]
```diff
@@ -21,17 +21,17 @@ cy:
       heading:
       success:
     your_account:
+      about_description_html:
```
[the remark this answers is not in the document]
Comment (KludgeKML) [reply config/locales/cy.yml:24]: It's both "fully alphabetize" and "normalise indentation" - which I think in this case is mainly correcting the confusing situation with arrays in yaml (ie that the keys of the array items are indented, but the - that indicates the start of a new array item is supposed to be inline with the key that starts the array, except that it also accepts indented dashes).

File: config/locales/en.yml [hunk 3]
```diff
@@ -481,73 +481,73 @@ en:
...
-      - title: Passports, travel and living&nbsp;abroad
-        description: Includes renewing passports and travel advice by country
-        link: /browse/abroad
-      - title: Visas and immigration
-        description: Apply to visit, work, study, settle or seek asylum in the&nbsp;UK
-        link: /browse/visas-immigration
-      - title: Working, jobs and&nbsp;pensions
-        description: Includes holidays, finding a job and redundancy
-        link: /browse/working
+    - title: Benefits
```
[the remark this answers is not in the document]
Comment (KludgeKML) [reply config/locales/en.yml:484]: Ah yes, see above.

File: lib/tasks/consolidation.rake [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    key_parts = key.split(".")
+    abort("You need to specify a key to copy") unless key
+
+    puts("Copying #{key} translations from government-frontend to frontend")
+
+    frontend_locale_files = Dir.entries(Rails.root.join("config/locales")).select { |name| name.end_with?(".yml") }.sort
+
+    frontend_locale_files.each do |flf|
+      government_frontend_keys = YAML.load(File.read(Rails.root.join("../government-frontend/config/locales", flf)))
+      locale_key_parts = [flf.gsub(".yml", "")] + key_parts
```
[the remark this answers is not in the document]
Comment (KludgeKML) [reply lib/tasks/consolidation.rake:19]: Yep

File: lib/tasks/consolidation.rake (hunk 1 above)
[the remark this answers is not in the document]
Comment (KludgeKML) [reply lib/tasks/consolidation.rake:5]: Hmm... I'm using zsh and I didn't need to escape the brackets. I know you need to put the whole thing in speechmarks if you're running it directly in kubernetes, but that's more about the colon I think. Let me try again locally and see what's what.

File: config/locales/no.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+---
+no:
```
[the remark this answers is not in the document]
Comment (KludgeKML) [reply config/locales/no.yml:2]: Ah, interesting. Added.

File: lib/tasks/consolidation.rake (hunk 1 above)
[the remark this answers is not in the document]
Comment (KludgeKML) [reply lib/tasks/consolidation.rake]: I've updated to add a bigger description, and have put the rake command in speechmarks, which is a standard way to get them to work in kubernetes and also works for zsh.

**Review by leenagupte — APPROVED** [review]
Thanks for updating the Usage description ⭐
